### DISCUSSION TITLE
Cannot run Junit5 test suite

### DISCUSSION THREAD

**Discussion by marc-ernst** [eclipse-tycho/tycho#1484] [Q&A]
Hi, I've tried a lot of different things to run a junit5 test suite with tycho surefire. In Eclipse the suite is executed fine but with tycho it does not find any tests. I can run the whole test suite with tycho and it recoginizes all single tests but when I try to run only the test suite it does not find any tests. I've tried with include and with testclass. I renamed my test suite to end with "*Test" or "*Tests" but nothing helped.

import org.junit.platform.suite.api.SelectClasses;
import org.junit.platform.suite.api.Suite;

import com.MyBasicTests;

@Suite
@SelectClasses({ MyBasicTests.class })
public class RunAllTest {

}

tychon version 2.7.5

      <plugin>
        <groupId>org.eclipse.tycho</groupId>
        <artifactId>tycho-surefire-plugin</artifactId>
        <version>${tycho-version}</version>
        <configuration>
          <useUIHarness>true</useUIHarness>
          <useUIThread>false</useUIThread>
          <application>${applicationName}</application>
          <argLine>
            ${tycho.testArgLine}
          </argLine>
          <includes>
            <include>**/*RunAllTest.java</include>
          </includes>
        </configuration>
      </plugin>

**Comment by mickaelistria** [comment]
Please try Tycho 3.0.0 or 3.1.0-SNAPSHOT.

**Comment by marc-ernst** [comment]
I'm having big problems with using 3.0.0:
1) Error in custom provider, java.lang.TypeNotPresentException: Type org.eclipse.tycho.p2maven.repository.EclipsePluginArtifactHandler not present
  at ClassRealm[extension>org.eclipse.tycho:tycho-maven-plugin:3.0.0, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@81438b0e] (via modules: org.eclipse.sisu.wire.WireModule -> org.eclipse.sisu.plexus.PlexusBindingModule)
  while locating org.apache.maven.artifact.handler.ArtifactHandler annotated with @com.google.inject.name.Named(value="eclipse-plugin")

**Reply by mickaelistria** [reply]
> I'm having big problems with using 3.0.0

So those problems should be reported and fixed. I don't think you'll find anyone willing to support 2.x for free now that 3.x is out.
One possible workaround for such can be to clear some bits of your m2 repo:
* org/eclipse/tycho
* .cache/tycho
* p2
* maybe even org/eclipse in general

and try again.

**Reply by marc-ernst** [reply]
I removed the whole repository folder but still the same issue. The first error message is:
WARNING] Error injecting: org.eclipse.tycho.p2maven.repository.EclipsePluginArtifactHandler
java.lang.TypeNotPresentException: Type org.eclipse.tycho.p2maven.repository.EclipsePluginArtifactHandler not present
    at org.eclipse.sisu.space.URLClassSpace.loadClass (URLClassSpace.java:147)
    at org.eclipse.sisu.space.NamedClass.load (NamedClass.java:46)
    at org.eclipse.sisu.space.AbstractDeferredClass.get (AbstractDeferredClass.java:48)
    at com.google.inject.internal.ProviderInternalFactory.provision (ProviderInternalFactory.java:85)

**Comment by mickaelistria** [comment]
Are you using Java 17 to run your build?

**Reply by marc-ernst** [reply]
Picked up _JAVA_OPTIONS: -Djava.net.preferIPv4Stack=true
openjdk version "17" 2021-09-14
OpenJDK Runtime Environment (build 17+35-2724)
OpenJDK 64-Bit Server VM (build 17+35-2724, mixed mode, sharing)

**Reply by mickaelistria** [reply]
OK, created https://github.com/eclipse-tycho/tycho/issues/1487 to track that. Please add any detail you think can help to that issue.

**Comment by marc-ernst** [comment]
Found the issue, I think you can close it. My JAVA_HOME was pointing to a different java then in the PATH, I'm currently trying to run the tests with 3.0.0

**Comment by marc-ernst** [comment]
No, still the same issue, doesn't find any tests.

**Comment by marc-ernst** [comment]
But runs all tests when I just use <include>**/*Test.java</include>

**Reply by laeubi** [reply]
Can you provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue?

**Comment by Phillipus** [comment]
I have the same problem. See https://github.com/eclipse-tycho/tycho/issues/2462
